api-reference/_hidden/dxDataGridColumn/calculateGroupValue.md

```diff
@@ -14,7 +14,21 @@ The calculated value used to create and sort groups.
 
 ---
 
-The **DataGrid** groups its rows by the grouping column. If you want to use custom grouping, change values of this column in the **calculateGroupValue** function. As a result, group cells' headers will display the changed values that may include extra information. To remove it, specify a custom template in the [groupCellTemplate](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#groupCellTemplate) option ([groupCellRender](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#groupCellRender) in React). Refer to this <a href="https://github.com/DevExpress-Examples/DataGrid---How-to-apply-custom-sorting-to-a-grouped-column" target="_blank">Github repository</a> for an example. 
+The following example shows how to group data by date relative to the current date:
+
+    calculateGroupValue: function(rowData){
+        let date = new Date(rowData.DateColumn);
+        if(date = isTodayDate()){
+            return "Today";
+        }
+        if(date = isYesterdayDate()){
+            return "Yesterday";
+        }
+        //...
+        return "Older";
+    }
+
+The **DataGrid** groups its rows by the grouping column. If you want to use custom grouping, change values of this column in the **calculateGroupValue** function. As a result, group cells' headers will display the changed values that may include extra information. To remove it, specify a custom template in the [groupCellTemplate](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#groupCellTemplate) property ([groupCellRender](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#groupCellRender) in React). Refer to this <a href="https://github.com/DevExpress-Examples/DataGrid---How-to-apply-custom-sorting-to-a-grouped-column" target="_blank">Github repository</a> for implementation.
```

Comment:
There are many things wrong with this topic. Short description is overloaded, full description lacks structure, the example doesn't target different frameworks and doesn't even work in jQuery. Let's rework everything and keep it simple.

Short description:
Calculates custom values used to group this column.

Full description (we will follow the structure of the calculateSortValue description):

This property accepts the name of the data source field that provides values by which to group this column...

%code_example%

... or a function that creates custom groups:

%code_example%

Group captions and appearance can be customized. To do this, define the [groupCellTemplate]. Refer to the following GitHub repository for an example: %url%

**calculateGroupValue** is also used to sort groups when the column is grouped.